src/test/fuzz/eval_script.cpp

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2009-2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <script/interpreter.h>
+#include <test/fuzz/fuzz.h>
+
+#include <limits>
+
+void test_one_input(const std::vector<uint8_t>& buffer)
+{
+    const CScript script(buffer.begin(), buffer.end());
+    for (const auto sig_version : {SigVersion::BASE, SigVersion::WITNESS_V0}) {
+        for (const auto flags : {0U, std::numeric_limits<unsigned int>::max()}) {
```

Comment:
Can you explain why the flags are hardcoded (as opposed to be read from the input)?

Reply:
not sure if this helps, but there is a `static bool IsValidFlagCombination` helper

Reply:
It is a trade-off: while reading the flags could increase coverage it would also greatly increase the search space in a way that is not trivial for a fuzzer to reduce intelligently (AFAICT). Based on my reading of how `flags` are affecting the code paths taken in `EvalScript` I came to the conclusion that `{0U, std::numeric_limits<unsigned int>::max()}` seemed like a good way to limit search space while retaining good coverage. Let me know if you can find counter examples of code paths that won't be taken due to this limitation that would warrant increasing the search space :)

Reply:
It might give you the same line coverage, but not the same path coverage. E.g. CLTV and CSV are always both enabled or both disabled, but never only one of them is enabled. Do you have any reading suggestions for me that explain your intuition that the fuzzer will conceptually have a harder time covering the search space when the input data is used to decide which code path to take?

Reply:
I'll try to find a link :)  I've now switched to reading `flags` from input using the excellent `FuzzedDataProvider` helper. I'm sure that will work out well too :)